### PR TITLE
[BE] Refactoring test execution and improving comments

### PR DESCRIPTION
**Pull request by ZainRizvi**
Sharing code between the code that handles test results in parallel vs serial mode.

Note that the original version of this code had an inconsistency between the two versions where it would execute `print_to_stderr(err_message)` on every test that ran in parallel, but for serial tests it would only invoke `print_to_stderr(err_message)` if `continue_on_error` was also specified.  By sharing code, this PR changes that behavior to be consistent between the two modes.

Also adding some comments.

### <samp>🤖 Generated by Copilot at 029342c</samp>

> _Sing, O Muse, of the skillful coder who refined_
> _The PyTorch testing script, `run_test.py`, and shined_
> _A light on its obscure logic, with docstrings and comments_
> _And made it run more smoothly, with better error contents_

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -1318,14 +1320,17 @@ def main():
     )
     os.makedirs(REPO_ROOT / "test" / "test-reports", exist_ok=True)
 
-    def success_callback(err_message):
+    def handle_error_messages(err_message):
         if err_message is None:
-            return True
```
Comment (ZainRizvi) [inline test/run_test.py:1323]: Note that nothing actually read the return value of this method before, so it's okay to invert it for better code ergonomics.

File: test/run_test.py [hunk 2]
```diff
@@ -1356,12 +1361,9 @@ def success_callback(err_message):
             if can_run_in_pytest(test):
                 options_clone.pytest = True
             err_message = run_test_module(test, test_directory, options_clone)
-            if err_message is None:
-                continue
-            failure_messages.append(err_message)
-            if not options_clone.continue_through_error:
+            test_failed = handle_error_messages(err_message)
+            if test_failed and not options.continue_through_error:
```
Comment (huydhn) [inline test/run_test.py:1365]: Do you also need to have `failure_messages.append(err_message)` here in the if condition before raising the RuntimeError?  Probably not as it looks like it's not used

**Review by huydhn — APPROVED** [review]

File: test/run_test.py (hunk 2 above)
Comment (ZainRizvi) [reply test/run_test.py:1365]: That bit happens inside `handle_error_messages`

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**